web/src/components/vis/VisTile.vue

```diff
@@ -11,28 +13,91 @@ export default {
     },
   },
 
+  data() {
+    return {
+      scales: [
+        { scaleFactor: 0.5, cssClass: 'span1' },
+        { scaleFactor: 1, cssClass: 'span2' },
+        { scaleFactor: 1.5, cssClass: 'span3' },
+        { scaleFactor: 2, cssClass: 'span4' },
+        { scaleFactor: 2.5, cssClass: 'span5' },
+      ],
+      scaleIndex: 1,
+    };
+  },
+
   computed: {
-    hasControls() {
-      return !!this.$slots.controls;
+    scaleClass() {
+      return this.scales[this.scaleIndex].cssClass;
+    },
+    scaleFactor() {
+      return this.scales[this.scaleIndex].scaleFactor;
+    },
+    scaleOptions() {
+      const f = format('.0%');
+      return this.scales.map((d, i) => ({ text: f(d.scaleFactor), value: i }));
     },
   },
 };
 </script>
 
 <template lang="pug">
-v-flex(shrink=1)
+v-flex(shrink=1, :class="scaleClass")
   .white.rounded.relative
     v-toolbar.primary.darken-3.top-rounded(dark, flat, dense)
       v-toolbar-title {{ title }}
       v-spacer
-      v-toolbar-items(v-if="hasControls")
+      v-toolbar-items
+        v-btn(@click="scaleIndex--", :disabled="scaleIndex === 0", icon)
+          v-icon {{ $vuetify.icons.magnifyMinus }}
+        v-select.scaleFactor(:value="scaleIndex", @change="scaleIndex = parseInt($event, 10)",
+            :items="scaleOptions", hide-details, item-text="text", item-value="value")
+        v-btn(@click="scaleIndex++", :disabled="scaleIndex === scales.length - 1", icon)
```

Comment:
Better not to rely on the Vue event loop to protect you from invalid states.  If the page or the machine is under heavy load, `:disabled` may not compute fast enough. 

Almost every project I've got right now has [this](https://stackoverflow.com/questions/11409895/whats-the-most-elegant-way-to-cap-a-number-to-a-segment) in the utilities.

Recommend:

```
script
 import { clamp } from 'utils';
  methods
    setScaleIndex(value) { this.scaleIndex = clamp(value, min, max); },
...
template
  v-btn(@click="setScaleIndex(scaleIndex +/- 1)", ...)
```

Reply:
To be clear, I don't use the Number prototype, I just take the number as the first argument.

Reply:
done